hoot-core/src/main/cpp/hoot/core/io/OsmApiChangeset.cpp

```diff
@@ -68,64 +68,82 @@ XmlChangeset::XmlChangeset(const QList<QString> &changesets)
     loadChangeset(*it);
 }
 
-void XmlChangeset::loadChangeset(const QString &changesetPath)
+void XmlChangeset::loadChangeset(const QString &changeset)
 {
-  /* <osmChange version="0.6" generator="acme osm editor">
-   *   <create|modify|delete>
-   *     <node|way|relation.../>
-   *     ...
-   *   </create|modify|delete>
-   *   ...
-   * </osmChange>
-   */
-  QFile file(changesetPath);
+  if (QFile::exists(changeset))
+    loadChangesetFile(changeset);
+  else
+    loadChangesetXml(changeset);
+}
```

Comment:
These changes all for a file path to be passed in to the `loadChangeset()` function as well as actual raw OSM XML changesets.  All of the loading changes are to accomplish this for testing.